### PR TITLE
Add auto sign-in after sign-up usage documentation

### PR DESCRIPTION
**Pull request by helgabalashova**
_Issue #, if available:_

_Description of changes:_

Adds auto sign in usage in sign up section
Adds clientMetadata and validationData usage for auto sign-in within advanced workflows section

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/auth/js/advanced.mdx [hunk 1]
```diff
@@ -381,6 +381,28 @@ Auth.signIn({
   .catch(err => console.log(err));
 ```
 
+### Auto Sign-in Lambda triggers
+
+If you have `autoSignIn` enabled, you can pass `validationData` and `clientMetadata` as properties for `AutoSignInOptions`. They are used
+ to implement additional validations for the `authenticateUser` function.
```
Comment (elorzafe) [inline src/fragments/lib/auth/js/advanced.mdx:387]: I will not menton `authenticateUser` here

**Review by offlineprogrammer — APPROVED** [review]

File: src/fragments/lib/auth/js/advanced.mdx [hunk 2]
```diff
@@ -381,6 +381,28 @@ Auth.signIn({
   .catch(err => console.log(err));
 ```
 
+### Auto Sign-in Lambda triggers
+
+If you have `autoSignIn` enabled, you can pass `validationData` and `clientMetadata` as properties for `AutoSignInOptions`. They are used
```
Comment (elorzafe) [inline src/fragments/lib/auth/js/advanced.mdx:386]: Looking the whole context I will mention here only the validation data piece.

File: src/fragments/lib/auth/js/advanced.mdx [hunk 3]
```diff
@@ -381,6 +381,28 @@ Auth.signIn({
...
+    username,
+    password,
+    attributes: {
+        email,          // optional
+        phone_number,   // optional - E.164 number convention
+        // other custom attributes 
+    },
+    autoSignIn: { // optional - enables auto sign in after sign up
+        enabled: true,
+        clientMetadata, // optional
```
Comment (elorzafe) [inline src/fragments/lib/auth/js/advanced.mdx:400]: I would update `Auth.signUp` on L420 that also is available for `autoSignIn`

**Review by elorzafe — CHANGES_REQUESTED** [review]
I would reorganize a little bit the advance section to match the separation between `clientMetadata` and `validationData`

**Review by elorzafe — APPROVED** [review]
Looks great! 

Thanks @helgabalashova 🌮 🎉

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

**Pull request merged**